runtime/scripts.vim

```diff
@@ -306,7 +306,9 @@ else
     set ft=virata
 
     " Strace
-  elseif s:line1 =~# '^\(\[pid \d\+\] \)\=[0-9:.]* *execve(' || s:line1 =~# '^__libc_start_main'
+  elseif s:line1 =~# '^\(\[pid \d\+\] \)\=[0-9:.]* *execve('
+    \ || s:line1 =~# '^__libc_start_main'
+    \ || s:line1 =~# 'execve(.*)\s\+= \d\+'
```

Comment:
> So please give a precise example (or show the strace generated logfile).

I apologize. I thought this was trivial to reproduce (hence the example in the commit message, which I could have sworn to have tested). However, seems like it isn't. Here is what I observe.
(Thanks for verifying that it works at your end, btw) :

```
[sfernandez@localhost ~]$ cat /etc/os-release 
NAME=Fedora
VERSION="25 (Twenty Five)"
ID=fedora
VERSION_ID=25
PRETTY_NAME="Fedora 25 (Twenty Five)"
...
[sfernandez@localhost ~]$ strace -ftvo pstree.log pstree > /dev/null
[sfernandez@localhost ~]$ head -1 pstree.log 
23038 15:12:08 execve("/usr/bin/pstree", ["pstree"], ... "_=/usr/bin/strace"]) = 0
```
And opening this doesn't set the ft. Surprisingly (for me) tho, when I did the following tho:
```
# note, unlike above, here stderr is captured, instead of using the -o flag,
# which results in pid not being in the output

[sfernandez@localhost ~]$ strace -ftv pstree 2> pstree_stderr.log > /dev/null
[sfernandez@localhost ~]$ head -1 pstree_stderr.log 
15:17:47 execve("/usr/bin/pstree", ["pstree"], ... "_=/usr/bin/strace"]) = 0
```
Sets the ft without the patch !

In either case (with `-o` or redirecting stderr), doing a strace without opts, generates:
```
[sfernandez@localhost ~]$ head -1 pstree_stderr_no_opts.log 
execve("/usr/bin/pstree", ["pstree"], 0x7ffd017a81b0 /* 63 vars */) = 0
[sfernandez@localhost ~]$ head -1 pstree_-o.log 
execve("/usr/bin/pstree", ["pstree"], 0x7ffc72fbb640 /* 63 vars */) = 0
[sfernandez@localhost ~]$ vi pstree_stderr_no_opts.log pstree_-o.log 
```
and this *does* indeed set the ft.

So, I imagine, I might've messed something up when trying to create a patch to fix the issue of ft not being set for the first example shown above. If I understand your explanation of the regex, I am now not sure why the part of the regex preceding the execve doesn't match in that instance.